win/main.cpp

```diff
@@ -59,9 +59,66 @@ int32_t  GetOptionint32_t(const char *pszOption);
     }
 #endif
 
-int APIENTRY _tWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPTSTR lpCmdLine, int nCmdShow)
+namespace
 {
-    UNREFERENCED_PARAMETER(hPrevInstance);
+	// Convenience functions so we can use main() while still getting the WinMain window and console.
+	HINSTANCE GetHInstance()
+	{
+		return static_cast<HINSTANCE>(GetModuleHandleW(nullptr));
+	}
+
+	LPTSTR GetLPCmdLine()
+	{
+		// The first argument is the program name. To allow it to have spaces, it can be surrounded by 
+		// quotes. We must track if the first argument is quoted since a space is also used to separate 
+		// each parameter. 
+		bool isQuoted = false;
+		const wchar_t space = L' ';
+		const wchar_t quote = L'\"';
+		const wchar_t nullTerminator = L'\0';
+
+		LPTSTR lpCmdLine = GetCommandLine();
+
+		// The lpCmdLine in a WinMain is the command line as a string excluding the program name. 
+		// Program names can be quoted to allow for space characters so we need to deal with that. 
+		while (*lpCmdLine <= space || isQuoted) {
```

Comment:
This should probably break on the null char, or else it'll overrun if there are no spaces. e.g.
`while (*lpCmdLine && (*lpCmdLine != space || isQuoted)) {`